data-validation/use-cases.md

```diff
@@ -2,15 +2,148 @@
+- The container at `<https://alice.example/phr/diagnostics/>` has a validator set that mandates all data within it conform to `<https://shapes.example/test-diagnostic.ttl>`.
+- Because the data written in `<https://alice.example/phr/diagnostics/081203.ttl>` corresponds to the SHACL shape at `<https://shapes.example/test-diagnostic.ttl>`, it is written successfully.
+
+#### Validating data when a resource has mixed data
+
+- Mary uses ExampleTask to manage her projects and their associated tasks.
+- She is working on a Supply Chain project. ExampleTask stores both the project metadata, and the associated tasks for it at `<https://alice.example/projects/supply-chain.ttl>`.
+- There are multiple validators set on `<https://alice.example/projects/supply-chain.ttl>`. Specifically, data written to this resource must conform to both `<https://shapes.example/project.shex>` and `<https://shapes.example/task.shex>`.
+
+#### Validating sets of data
+
+- Rob uses PondStat to record the temperature of his pond.
+- PondStat takes one thousand measurements per second and stores the result to Rob's pod, because Rob is serious about fluctuations in pond climate.
+- This data is stored in a container in his pod at `<https://rob.example/pond/measurement/>`.
+- Since it would be expensive to send 1000 requests per second, PondStat sends one request per second containing 1000 instances of data that conforms to the `<https://shapes.example/pond-measurement.ttl>` SHACL shape.
+- Because each instance of data written conforms to the SHACL shape at `<https://shapes.example/pond-measurement.ttl>`, it is written successfully.
+
+### Modifying shape validation
+
+These use cases are focused on *modifying* validations that are already in place on resources or containers.
+
+#### Adding additional shape validations to a resource
+
+- Alice uses FooHealth to manage her Personal Health Data.
+- Alice's doctor uses an EHR system called BarEHR that is Solid compatible.
+- BarEHR maintains a care plan resource at `<https://alice.example/phr/care-plan.ttl>` that maps to a number of other resources such as prescribed diagnostics, prescriptions for medication, and other required activities.
+- This care plan resource has a validator set that ensures it conforms to `<https://shapes.example/care-plan.shex>`.
+- A new version of BarEHR allows it to canonicalize and sign the care plan data, then store the results in the same resource, conforming to the `<https://shapes.example/data-signature.shex>` shape.
+- BarEHR adds `<https://shapes.example/data-signature.shex>` to the allowed validators for `<https://alice.example/phr/care-plan.ttl>`, such that both must be present (the care plan is always signed) to pass validation.
+
+#### Adding additional shape validations to a container
+
+- Bob uses FooChat to communicate with his co-workers.
+- Bob creates a chatroom where he can talk to his colleague Rob.
```

Comment:
Suggestion not specific to this PR, same use case will apply to other projects than validation, we may need some base use case and then project specific aspects of the same use case. Direct chat can have variants where chatroom gets created in organization storage or one or both of personal storage. It can also have requirements like data redundancy eg. when it gets created in Bob's personal storage, Rob wants to have copy of all messages in his storage for redundancy - Bob's storage going offline shouldn't cause Rob to loose chat history. I see it useful to have all chat related use cases organized together and then explore different aspects of them in various projects.

Reply:
Agree that we should have top level use cases for Solid. Will propose as a broader issue outside of this PR.